### PR TITLE
Correct nodejs packagegen

### PR DESCRIPTION
**Pull request by tbetbetbe**
- fixes issues with the nodejs packagegen
- updates it so that it creates commonjs module rather than a simple json file

### CODE REVIEW

**Comment by saicheems** [comment]
Use the latest version of protobufjs commit title ends in a period and shouldn't

File: config/dependencies.yml [hunk 1]
```diff
@@ -6,6 +6,8 @@ auth:
     version: '1.4.11'
   ruby:
     version: '0.4.1'
+  nodejs:
+    version: '0.9.7'
```
Comment (saicheems) [inline config/dependencies.yml]: Pretty old version right? I think it's on v5 or so now.

File: lib/api_repo.js [hunk 2]
```diff
@@ -591,21 +600,34 @@ ApiRepo.prototype._makeProtocFunc = function _makeProtocFunc(opts, language) {
 /**
  * Helps construct a JSON representation each proto file for the nodejs build.
  *
- * The JSON representation is constructed via a opts.pbBuilder, a
- * ProtoBuf.Builder
- *
  * @param {object} opts provides configuration info
- * @param {object} opts.pbBuilder a ProtoBuf.Builder instance
+ * @param {object} opts.root the virtual root folder where all protos
```
Comment (saicheems) [inline lib/api_repo.js]: Seems an incomplete sentence

**Comment by saicheems** [comment]
First pass looks good

File: config/dependencies.yml (hunk 1 above)
Comment (tbetbetbe) [inline config/dependencies.yml]: this is not nodejs itself, but of google-auth-library-nodejs

File: lib/api_repo.js (hunk 2 above)
Comment (tbetbetbe) [inline lib/api_repo.js]: Fixed

**Comment by tbetbetbe** [comment]
> "Use the latest version of protobufjs" commit title ends in a period and shouldn't
I squashed this commit

**Comment by saicheems** [comment]
Also believe the top commit should be "Correct a comment"
Sorry to be such a stickler :P

Everything else lgtm

**Comment by tbetbetbe** [comment]
> Also believe the top commit should be "Correct a comment"
Fixed :)

**Pull request merged**